cmd/chisel/cmd_info_test.go

```diff
@@ -0,0 +1,283 @@
+	input:   sampleRelease,
+	query:   []string{"foo", "mypkg_foo", "bar_foo"},
+	stdout: `
+package: mypkg
+archive: ubuntu
+slices:
+    foo:
+        contents:
+            /etc/foo: {}
+            /etc/foo-dir/:
+                make: true
+                mode: 0644
+`,
+	err: ".*no slice definitions found for: .*foo.*bar_foo.*",
+}, {
+	summary: "No args",
+	input:   sampleRelease,
+	err:     ".*required argument.*not provided.*",
+}, {
+	summary: "Empty, whitespace args",
+	input:   sampleRelease,
+	query:   []string{"", "    "},
+	err:     `no slice definitions found for: "", "    "`,
+}, {
+	summary: "Bad format slices",
+	input:   sampleRelease,
+	query:   []string{"foo_bar_foo", "a_b", "7_c", "a_b c", "a_b x_y"},
+	err:     ".*no slice definitions found for:.*",
+}}
+
+const defaultChiselYaml = `
+	format: chisel-v1
+	archives:
+		ubuntu:
+			version: 22.04
+			components: [main, universe]
+`
+
+var sampleRelease = map[string]string{
+	"chisel.yaml": string(defaultChiselYaml),
+	"slices/mypkg.yaml": `
+		package: mypkg
+
+		slices:
+			foo:
+				contents:
+					/etc/foo:
+					/etc/foo-dir/: {make: true, mode: 0644}
+			bar:
+				essential:
+					- mypkg_foo
+				contents:
+					/etc/bar.conf: {text: TODO, mutable: true, arch: riscv64}
+					/lib/*-linux-*/bar.so: {arch: [amd64,arm64,i386]}
+					/usr/bin/bar: {symlink: /bin/bar}
+					/usr/bin/baz: {copy: /bin/bar}
+					/usr/lib/bar*.so:
+					/usr/share/bar/*.conf: {until: mutate}
+					/bin/bar:
+				mutate: |
+					dir = "/usr/share/bar/"
+					conf = [content.read(dir + path) for path in content.list(dir)]
+					content.write("/etc/bar.conf", "".join(conf))
+			baz:
+				essential:
+					- libpkg_libs
+					- mypkg_foo
+					- mypkg_bar
+	`,
+	"slices/libpkg.yaml": `
+		package: libpkg
+
+		slices:
+			libs:
+				contents:
+					/usr/lib/libpkg.so*:
+	`,
+}
+
+func (s *ChiselSuite) TestInfoCommand(c *C) {
+	for _, test := range infoTests {
+		c.Logf("Summary: %s", test.summary)
+
+		s.ResetStdStreams()
+
+		dir := c.MkDir()
+		for path, data := range test.input {
+			fpath := filepath.Join(dir, path)
+			err := os.MkdirAll(filepath.Dir(fpath), 0755)
+			c.Assert(err, IsNil)
+			err = os.WriteFile(fpath, testutil.Reindent(data), 0644)
+			c.Assert(err, IsNil)
+		}
+
+		prefix := []string{"info"}
+		if len(test.query) > 0 {
+			prefix = append(prefix, "--release", dir)
+		}
+		test.query = append(prefix, test.query...)
+
+		_, err := chisel.Parser().ParseArgs(test.query)
+		if test.err != "" {
+			c.Assert(err, ErrorMatches, test.err)
+			continue
+		} else {
+			c.Assert(err, IsNil)
+		}
+
+		// parse the expected and output YAMLs and compare
+		var testYAML interface{}
+		dec := yaml.NewDecoder(bytes.NewBuffer([]byte(test.stdout)))
+		err = dec.Decode(&testYAML)
+		c.Assert(err, IsNil)
+		var outputYAML interface{}
+		dec = yaml.NewDecoder(bytes.NewBuffer([]byte(s.Stdout())))
+		err = dec.Decode(&outputYAML)
+		c.Assert(err, IsNil)
+		c.Assert(testYAML, DeepEquals, outputYAML)
```

Comment:
Change this to check the final output:
```go
		c.Assert(strings.TrimSpace(s.Stdout()), DeepEquals, strings.TrimSpace(test.stdout))
```

Reply:
Thanks, should have done this in the first place.

Reply:
It was my bad probably. Months ago I was under the impression that we did things differently for testing, I try to keep it as simple as possible now :).